### PR TITLE
Feat/case insensitive in operators

### PR DESCRIPTION
**Pull request by andreas-unleash**
Client specification test for case insensitive IN and NOT_IN operators

[Project](https://linear.app/unleash/project/[low][none]-case-insensitive-in-and-not-in-operators-161256fdb9e3)
## About the changes

Closes #

### CODE REVIEW

File: specifications/16-case-insensitive-in-operators.json [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+                ]
+              }
+            ],
+            "segments": [1]
+          }
+        ]
+      },
+      {
+        "name": "F9.globalSegmentAndConstraint_IN_inverted",
+        "description": "With global segment and constraint both on",
```
Comment (sighphyre) [inline specifications/16-case-insensitive-in-operators.json:31]: The description here seems to be the same for each toggle, can we give these a more unique name describing what they do? They used in some of the SDK test scaffolds to report which test failed so it's useful if they're unique and descriptive

File: specifications/16-case-insensitive-in-operators.json [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+                "contextName": "appName",
+                "operator": "IN",
+                "caseInsensitive": true,
+                "inverted": false,
+                "values": [
+                  "app"
+                ]
+              }
+            ],
+            "segments": [1]
```
Comment (sighphyre) [inline specifications/16-case-insensitive-in-operators.json:25]: I don't think the segments add much value to these tests. They're just constraints in a dictionary and they add a lot of cognitive complexity to understanding this suite. They're already quite well covered in other suites so I would suggest either removing them or only including a single test with a segment

**Review by sighphyre — CHANGES_REQUESTED** [review]
I would like to see two things here before we pull this in:
- Some more tests that exercise the case insensitivity here, right now everything is checking "App" vs "app". Can we also include some tests that check mixed cases, special characters and substrings
- Some negative case tests would also be nice here. What happens when there's no value in the context for example?

File: specifications/16-case-insensitive-in-operators.json (hunk 2 above)
Comment (andreas-unleash) [reply specifications/16-case-insensitive-in-operators.json:25]: Agreed

File: specifications/16-case-insensitive-in-operators.json (hunk 1 above)
Comment (andreas-unleash) [reply specifications/16-case-insensitive-in-operators.json:31]: Thanks makes sense

File: specifications/16-case-insensitive-in-operators.json [hunk 3]
```diff
@@ -0,0 +1,163 @@
+{
+  "name": "16-case-insensitive-in-operators",
+  "state": {
+    "version": 2,
+    "features": [
+      {
+        "name": "F9.globalSegmentAndConstraint_IN",
```
Comment (sighphyre) [inline specifications/16-case-insensitive-in-operators.json:7]: Last thing, the names don't make sense anymore now that we've removed the segments. If you patch the names here I'm happy to approve

**Review by sighphyre — CHANGES_REQUESTED** [review]

File: specifications/16-case-insensitive-in-operators.json (hunk 3 above)
Comment (andreas-unleash) [reply specifications/16-case-insensitive-in-operators.json]: 👍

**Pull request merged**